Azure/connectors/azure_blob.py

```diff
@@ -0,0 +1,197 @@
+        """Init AzureBlobConnector."""
+
+        super().__init__(*args, **kwargs)
+        self.context = PersistentJSON("context.json", self._data_path)
+
+    @property
+    def last_event_date(self) -> datetime:
+        """
+        Get last event date.
+
+        Returns:
+            datetime:
+        """
+        now = datetime.now(timezone.utc)
+        one_hour_ago = (now - timedelta(hours=1)).replace(microsecond=0)
+
+        with self.context as cache:
+            last_event_date_str = cache.get("last_event_date")
+
+            # If undefined, retrieve events from the last 1 hour
+            if last_event_date_str is None:
+                return one_hour_ago
+
+            # Parse the most recent date seen
+            last_event_date = isoparse(last_event_date_str).replace(microsecond=0)
+
+            # We don't retrieve messages older than 1 hour
+            if last_event_date < one_hour_ago:
+                return one_hour_ago
+
+            return last_event_date
+
+    def azure_blob_wrapper(self) -> AzureBlobStorageWrapper:
+        """
+        Get Azure blob wrapper.
+
+        Returns:
+            AzureBlobStorageWrapper:
+        """
+        if not self._azure_blob_storage_wrapper:
+            config = AzureBlobStorageConfig(**self.configuration.dict(exclude_unset=True, exclude_none=True))
+            self._azure_blob_storage_wrapper = AzureBlobStorageWrapper(config)
+
+        return self._azure_blob_storage_wrapper
+
+    async def get_azure_blob_data(self) -> list[str]:
+        """
+        Get Azure Blob Storage data.
+
+        Returns:
+            list[str]:
+        """
+        blob_list = self.azure_blob_wrapper().list_blobs()
+        _last_modified_date = self.last_event_date
+        records: list[Any] = []
+        async for blob in blob_list:
+            if blob.last_modified > self.last_event_date:
+                if _last_modified_date is None or blob.last_modified > _last_modified_date:
+                    _last_modified_date = blob.last_modified
+
+                file, content = await self.azure_blob_wrapper().download_blob(blob.name, download=True)
+
+                if file:
+                    async with aiofiles.open(file, "rb") as file_data:
+                        file_content = await file_data.read()
+
+                        if is_gzip_compressed(file_content):
+                            file_content = decompress(file_content)
+
+                        records.extend(
+                            self.filter_blob_data(orjson.loads(file_content.decode("utf-8")), self.last_event_date)
+                        )
+
+                    await delete_file(file)
+
+                if content:
+                    records.extend(self.filter_blob_data(orjson.loads(content), self.last_event_date))
+
+        result: list[str] = await self.push_data_to_intakes(
+            [orjson.dumps(event).decode("utf-8") for event in records],
+        )
```

Comment:
according to the previous comment:
```suggestion
        result: list[str] = await self.push_data_to_intakes(events=records)
```